### ISSUE TITLE
Gallica OPDS doesn't always provide item author

### ISSUE THREAD

**Issue by Frenzie** [koreader/koreader#6418]
* KOReader version: 2020.07
* Device: all

#### Issue
The code depends on item.author being available, this doesn't seem to be guaranteed.

```
./luajit: frontend/ui/widget/opdsbrowser.lua:547: attempt to concatenate field 'author' (a nil value)
stack traceback:
	frontend/ui/widget/opdsbrowser.lua:547: in function 'downloadFile'
	frontend/ui/widget/opdsbrowser.lua:650: in function 'callback'
	frontend/ui/widget/button.lua:239: in function 'action'
	frontend/ui/uimanager.lua:821: in function '_checkTasks'
	frontend/ui/uimanager.lua:1141: in function 'handleInput'
	frontend/ui/uimanager.lua:1227: in function 'run'
	./reader.lua:299: in main chunk
	[C]: at 0x563ba7d07791
```

```
07/22/20-18:42:06 DEBUG {
    ["image"] = "http://gallica.bnf.fr/ark:/12148/bpt6k6531118k.highres",
    ["thumbnail"] = "http://gallica.bnf.fr/ark:/12148/bpt6k6531118k.thumbnail",
    ["id"] = "http://gallica.bnf.fr/ark:/12148/bpt6k6531118k",
    ["acquisitions"] = {
        [1] = {
            ["type"] = "application/epub+zip",
            ["href"] = "http://gallica.bnf.fr/ark:/12148/bpt6k6531118k.epub"
        }
    },
    ["updated"] = "2013-08-01T00:00:00",
    ["title"] = "L'escoufle : roman d'aventure / publ. pour la première fois d'après le manuscrit unique de l'Arsenal par H. Michelant et P. Meyer",
    ["text"] = "L'escoufle : roman d'aventure / publ. pour la première fois d'après le manuscrit unique de l'Arsenal par H. Michelant et P. Meyer"
}
```

I came across this while looking into #5997. I'll fix it at the same time.

#### Steps to reproduce
Download something from Gallica OPDS.

**Comment by hius07** [comment]
> bpt6k6531118k.epub

Well, this name is not very good to save. Can we have a choice: use metadata or keep original?

**Comment by hius07** [comment]
Loading two versions of a book from Flibusta (e.g. by different translators) we will get the same metadata author+title filename.

**Comment by Frenzie** [comment]
Technically it's slightly different in content-disposition:

```
$ curl -I https://gallica.bnf.fr/ark:/12148/bpt6k6531118k.epub

HTTP/2 200 
date: Wed, 22 Jul 2020 17:51:58 GMT
server: Apache
set-cookie: JSESSIONID=C58C4D4FC4F4EC420585C14713631478; Path=/; Secure; HttpOnly
content-disposition: inline;filename="L'escoufle_:_roman_d'aventure_/_[...]_bpt6k6531118k.epub"
content-type: application/epub+zip;charset=UTF-8
content-language: fr-FR
content-length: 1311858
vary: User-Agent
```

But yes, this is how it is on most OPDS servers afaict.

**Comment by Frenzie** [comment]
Fixed by #7384.
